### DISCUSSION TITLE
Does `Vector2LineAngle` need normalized vectors?

### DISCUSSION THREAD

**Discussion by couryrr** [gen2brain/raylib-go#397] [Q&A]
I have been doing some simple visualization while trying to re-learn linear algebra. While doing so I was trying to show the angle from the center of the screen to the mouse position .

I was passing normalized vectors based on  the comment [here](https://github.com/gen2brain/raylib-go/blob/62016ee92fc0d1c8c2803f1a8a4e82124b92ba55/raylib/raymath.go#L111):

`angle := rl.Vector2LineAngle(rl.Vector2Normalize(mid), rl.Vector2Normalize(mouse))` 

Since that did not give me the value I was expecting  I updated my code to use `Vector2Angle` but missed updating the method so had:

`angle := rl.Vector2LineAngle(mid, mouse)` 

The above call does not use normalized vectors but give me the correct degree.

Here is the simple code I put together:

Curious too how to use `Vector2Angle` the degree never seems to match what I expect with the actual vector or the normalized.

```go

package main

import (
	"fmt"

	rl "github.com/gen2brain/raylib-go/raylib"
)

func main(){
    rl.InitWindow(1920, 1080, "Learning raylib math")
    defer rl.CloseWindow()
    
    mid := rl.Vector2{X: 1920/2, Y:1080/2}

    for !rl.WindowShouldClose(){
        mouse := rl.GetMousePosition()
        rl.BeginDrawing()

        rl.ClearBackground(rl.Black)
        rl.DrawPixelV(mid, rl.Green)
        rl.DrawCircleLines(int32(mid.X), int32(mid.Y), 50, rl.Green)
        
        rl.DrawLineV(mid, mouse, rl.Red)
        
        angle := rl.Vector2Angle(mid, mouse)
        lineAngle := rl.Vector2LineAngle(mid, mouse)

        rl.DrawText(fmt.Sprintf("Angle:\nDegree: %f\nLine Angle:\nDegree: %f\nVectors:\nMid: %v - %v\nMouse: %v - %v", angle * rl.Rad2deg,lineAngle*rl.Rad2deg, mid, rl.Vector2Normalize(mid), mouse, rl.Vector2Normalize(mouse)), 10, 10, 20, rl.Green)
        rl.EndDrawing()
    }

}

```

**Comment by JupiterRider** [comment]
Vector2LineAngle and Vector2Angle are producing the same result like in raylib C.
At least our Go functions are not faulty.
